jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -380,22 +416,41 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
         except ValueError:  # Clearer error message
             raise ValueError('Missing or invalid background value')
 
+        if self.config == "cubeviz" and data.ndim > 2:
+            cube_slice_plg = self.app._jdaviz_helper.plugins["Slice"]._obj
+            cube_slice = int(cube_slice_plg.slice)
+            comp_data = comp.data[:, :, cube_slice].T  # nx, ny --> ny, nx
+            # Similar to coords_info logic.
+            if '_orig_spec' in getattr(data, 'meta', {}):
+                w = data.meta['_orig_spec'].wcs
+            else:
+                w = data.coords
+        else:  # "imviz"
+            comp_data = comp.data  # ny, nx
+            w = data.coords
+
         if hasattr(reg, 'to_pixel'):
             sky_center = reg.center
-            xcenter, ycenter = data.coords.world_to_pixel(sky_center)
+            if self.config == "cubeviz" and data.ndim > 2:
+                ycenter, xcenter = w.world_to_pixel(

```

Comment:
Note: I tried to be lazy here, i.e., extracting just the `wcs.celestial` component and then use it without wavelength input, but the answer came out wrong, probably because of the specutils transpose, so let's just do it the hard way here.